packages/components/src/LightBox/LightBox.tsx

```diff
@@ -43,43 +42,90 @@ interface LightBoxProps {
   onRequestClose(options: RequestCloseOptions): void;
 }
 
+const swipeConfidenceThreshold = 10000;
+const swipePower = (offset: number, velocity: number) => {
+  return Math.abs(offset) * velocity;
+};
+
+const variants = {
+  enter: (direction: number) => {
+    return {
+      x: direction > 0 ? 2000 : -2000,
+      opacity: 0,
+    };
+  },
+  center: {
+    zIndex: 1,
+    x: 0,
+    opacity: 1,
+  },
+};
+
+const imageTransition = {
+  x: { type: "spring", stiffness: 300, damping: 30 },
+  opacity: { duration: 0.2 },
+};
+
 export function LightBox({
   open,
   images,
   imageIndex = 0,
   onRequestClose,
 }: LightBoxProps) {
   const [currentImageIndex, setCurrentImageIndex] = useState(imageIndex);
-  let nextSrc = undefined;
-  let prevSrc = undefined;
-  if (images.length > 1) {
-    nextSrc = images[(currentImageIndex + 1) % images.length].url;
-    prevSrc =
-      images[(currentImageIndex + images.length - 1) % images.length].url;
-  }
+  useRefocusOnActivator(open);
+
+  useOnKeyDown(() => {
+    onRequestClose({ lastPosition: currentImageIndex });
+  }, "Escape");
+
+  useOnKeyDown(handleMovePrevious, {
+    key: "ArrowLeft",
+  });
+
+  useOnKeyDown(handleMoveNext, {
+    key: "ArrowRight",
+  });
 
   useEffect(() => {
     setCurrentImageIndex(imageIndex);
-  }, [imageIndex]);
+  }, [imageIndex, open]);
 
   return (
     <>
-      {open && (
-        <ExternalLightBox
-          wrapperClassName={styles.wrapper}
-          mainSrc={images[currentImageIndex].url}
-          enableZoom={false}
-          nextSrc={nextSrc}
-          prevSrc={prevSrc}
-          imageTitle={images[currentImageIndex].title}
-          imageCaption={images[currentImageIndex].caption}
-          onCloseRequest={() =>
-            onRequestClose({ lastPosition: currentImageIndex })
-          }
-          onMovePrevRequest={handleMovePrevious}
-          onMoveNextRequest={handleMoveNext}
-        />
-      )}
+      <AnimatePresence initial={false}>
+        {open && (
+          <LightBoxWrapper key="lightbox">
+            <Toolbar>
+              <Title>{images[currentImageIndex].title}</Title>
+              <ButtonDismiss
+                ariaLabel="Close lightbox"
+                onClick={() => {
+                  onRequestClose({ lastPosition: currentImageIndex });
+                }}
```

Comment:
As a best practice, it's much better to put this in a function somewhere and reference it here as `onClick={handleClick}` since named functions are much more performant than anonymous